x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/SubstituteSurrogates.java

```diff
@@ -70,6 +71,10 @@ protected LogicalPlan rule(Aggregate aggregate) {
                 if (s instanceof AggregateFunction == false) {
                     // 1. collect all aggregate functions from the expression
                     var surrogateWithRefs = s.transformUp(AggregateFunction.class, af -> {
+                        // TODO: more generic than this?
```

Comment:
maybe in the future we can have a common interface for time series aggregate functions?

Reply:
Yes, will do this in a follow-up. Maybe NestedAggregateFunction.